src/main/java/org/powerbot/script/rt4/CollectionBox.java

```diff
@@ -0,0 +1,91 @@
+package org.powerbot.script.rt4;
+
+import org.powerbot.script.Condition;
+
+import static org.powerbot.script.rt4.Constants.*;
+
+public class CollectionBox extends ClientAccessor {
+
+	public CollectionBox(ClientContext ctx) {
+		super(ctx);
+	}
+
+	public boolean collectOffer(CollectionSlot slot, boolean toBank) {
+		if (!collectCoins(slot, toBank)) {
+			return false;
+		}
+		return collectItems(slot, toBank);
+	}
+
+	public boolean collectCoins(CollectionSlot slot, boolean toBank) {
+		Component item = slot.getComponent().component(4);
+		if (toBank) {
+			item.interact("Bank");
+		} else {
+			item.click();
+		}
+		return Condition.wait(() -> !item.visible(), 500, 5);
+	}
+
+	public boolean collectItems(CollectionSlot slot, boolean toBank) {
+		Component item = slot.getComponent().component(3);
+		if (toBank) {
+			item.interact("Bank");
+		} else {
+			item.click();
+		}
+		return Condition.wait(() -> !item.visible(), 500, 5);
+	}
+
+	public boolean open() {
+		if (opened()) {
+			return true;
+		}
+
+		Npc npc = ctx.npcs.toStream().action("Collect").nearest().first();
+		npc.interact("Collect");
```

Comment:
For all these interaction methods maybe do the following

```
if (npc.interact("Collect")) {
    return Condition.wait...
}
return false;
```

Reply:
went through and changed all interactions I could find to match that style